3_javascript/javascript2/week1/script.js

```diff
@@ -0,0 +1,128 @@
+//Find and count the Danish letters
+
+function countDanishLetters(inputString) {
+  const danishLetters = {
+    æ: 0,
+    ø: 0,
+    å: 0,
+  };
+
+  const lowercaseString = inputString.toLowerCase();
+
+  for (const char of lowercaseString) {
+    if (danishLetters.hasOwnProperty(char)) {
+      danishLetters[char]++;
+    }
+  }
+
+  const total = Object.values(danishLetters).reduce(
+    (acc, count) => acc + count,
+    0
+  );
+
+  const result = { total, ...danishLetters };
+
+  return result;
+}
+
+const danishString = "Jeg har en blå bil";
+const result1 = countDanishLetters(danishString);
+console.log(result1);
+
+const danishString2 = "Blå grød med røde bær";
+const result2 = countDanishLetters(danishString2);
+console.log(result2);
+
+//Spirit animal name generator
+
+document.addEventListener("DOMContentLoaded", function () {
+  const generateButton = document.getElementById("generateButton");
+  const spiritAnimalElement = document.getElementById("spiritAnimal");
+  const newAnimalButton = document.getElementById("newAnimalButton");
+  const eventSelector = document.getElementById("eventSelector");
+  const generateEventButton = document.getElementById("generateEventButton");
+
+  generateButton.addEventListener("click", function () {
+    generateSpiritAnimal();
+  });
```

Comment:
if you have your function declared somewhere else already, you don't need to write callback like this. Just writing:
```
generateButton.addEventListener('click', generateSpiritAnimal)
```
would be enough. 

I would like to talk through it on the next class, please remind me if I forget to explain this 🙏